### PR TITLE
Support for GetPurchaseHistory

### PR DESCRIPTION
**Pull request by syndarin**
New API for BillingProcessor:
isRequestBillingHistorySupported - to check if request purchase history supported
getPurchaseHistory - to request purchase history

### CODE REVIEW

File: library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java [hunk 1]
```diff
@@ -489,6 +490,42 @@ public boolean isOneTimePurchaseWithExtraParamsSupported(Bundle extraParams)
...
+	 * @return {@code true} if feature supported {@code false} otherwise
+	 */
+	public boolean isRequestBillingHistorySupported(String type) throws BillingCommunicationException
+	{
+		if (!type.equals(Constants.PRODUCT_TYPE_MANAGED) && !type.equals(Constants.PRODUCT_TYPE_SUBSCRIPTION))
+		{
+			throw new RuntimeException("Unsupported type " + type);
+		}
+
+		IInAppBillingService billing = billingService;
```
Comment (serggl) [inline library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: do we really need a local var here?

File: library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java [hunk 2]
```diff
@@ -489,6 +490,42 @@ public boolean isOneTimePurchaseWithExtraParamsSupported(Bundle extraParams)
 		return isOneTimePurchaseExtraParamsSupported;
 	}
 
+	/**
+	 * Checks does API support version 6 required to request purchase history
```
Comment (serggl) [inline library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: lets change this descritons to `Checks if API supports version 6 which required to request purchase history`.

 I ❤️ the descriptions in this PR BTW

File: library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java (hunk 1 above)
Comment (syndarin) [reply library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: Yes, `this.billingService` becomes `null` if service disconnects.

File: library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java (hunk 2 above)
Comment (syndarin) [reply library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java]: Updated

File: README.md [hunk 3]
```diff
@@ -238,6 +238,25 @@ public final Date purchaseTime;
 public final PurchaseInfo purchaseInfo;
 ```
 
+## Getting Purchase History
```
Comment (serggl) [inline README.md]: ❤️

**Comment by syndarin** [comment]
@serggl Thank you for accepting & merging PR. Can we expect updated artifact soon? (1.0.46 I suppose)

**Comment by serggl** [comment]
1.1.0 was just deployed

**Comment by jonboyandbailey472** [comment]
Good work yall

**Pull request merged**